internal/manifest/readycheck/deployment_ready_check.go

```diff
@@ -0,0 +1,59 @@
+package readycheck
+
+import (
+	apiappsv1 "k8s.io/api/apps/v1"
+	apicorev1 "k8s.io/api/core/v1"
+	"k8s.io/kubectl/pkg/util/deployment"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+const (
+	NewRSAvailableReason    = "NewReplicaSetAvailable"
+	FoundNewRSReason        = "FoundNewReplicaSet"
+	NewReplicaSetReason     = "NewReplicaSetCreated"
+	ReplicaSetUpdatedReason = "ReplicaSetUpdated"
+)
+
+// NewDeploymentReadyCheck creates a readiness check that verifies if a Deployment is ready.
+func NewDeploymentReadyCheck() *DeploymentReadyCheck {
+	return &DeploymentReadyCheck{}
+}
+
+type DeploymentReadyCheck struct{}
+
+func (c *DeploymentReadyCheck) Run(
+	deploy *apiappsv1.Deployment,
+) (shared.State, error) {
+	deploymentState := GetDeploymentState(deploy)
+	return deploymentState, nil
+}
+
+func GetDeploymentState(deploy *apiappsv1.Deployment) shared.State {
```

Comment:
I think this should not be an exported func (right now it is used in the test and Run only). In my opinion we can move this directly up into the Run (may be renamed) and test against the Run function.